optimum/exporters/features.py

```diff
@@ -0,0 +1,746 @@
+        """
+        Retrieves a model from its name and the feature to be enabled.
+
+        Args:
+            feature (`str`):
+                The feature required.
+            model (`str`):
+                The name of the model to export.
+            framework (`str`, *optional*):
+                The framework to use for the export. See `FeaturesManager.determine_framework` for the priority should
+                none be provided.
+            cache_dir (`str`, *optional*):
+                Path to a directory in which a downloaded pretrained model weights have been cached if the standard cache should not be used.
+
+        Returns:
+            The instance of the model.
+
+        """
+        framework = FeaturesManager.determine_framework(model, framework)
+        model_class = FeaturesManager.get_model_class_for_feature(feature, framework)
+        try:
+            model = model_class.from_pretrained(model, cache_dir=cache_dir)
+        except OSError:
+            if framework == "pt":
+                logger.info("Loading TensorFlow model in PyTorch before exporting.")
+                model = model_class.from_pretrained(model, from_tf=True, cache_dir=cache_dir)
+            else:
+                logger.info("Loading PyTorch model in TensorFlow before exporting.")
+                model = model_class.from_pretrained(model, from_pt=True, cache_dir=cache_dir)
+        return model
+
+    @staticmethod
+    def get_exporter_config_constructor(
+        model: Union["PreTrainedModel", "TFPreTrainedModel"], exporter: str, feature: str = "default"
+    ) -> ExportConfigConstructor:
+        """
+        Retrieves the proper exporter config constructor for the given model, feature and exporter backend.
+
+        Args:
+            model (`Union[PreTrainedModel, TFPreTrainedModel]`):
+                The model to export.
+            exporter (`str`):
+                The exporter to use.
+            feature (`str`, *optional*, detaults to `"default"`):
+                The name of the feature to check if it is available.
+
+        Returns:
+             `Tuple[str, ExportConfigConstructor]`: The model type as well as the `ExportConfig`
+             constructor for the requested backend.
+
+        """
+        model_type = model.config.model_type.replace("_", "-")
+        model_name = getattr(model, "name", "")
+        model_features = FeaturesManager.get_supported_features_for_model_type(
+            model_type, exporter, model_name=model_name
+        )
+        if feature not in model_features:
+            raise ValueError(
+                f"{model.config.model_type} doesn't support feature {feature} for the {exporter} backend."
+                f" Supported values are: {model_features}"
+            )
+
+        return FeaturesManager._SUPPORTED_MODEL_TYPE[model_type][exporter][feature]
+
+    def get_config(model_type: str, exporter: str, feature: str) -> ExportConfigConstructor:
```

Comment:
Is this method used anywhere? It looks quite redundant with the method right above

Reply:
Refactored! Thansk for the suggestion :+1: